### DISCUSSION TITLE
Question Regarding Multi-part Message.

### DISCUSSION THREAD

**Discussion by jaldeepkaneria** [goToMain/libosdp#172] [Q&A] [answered]
Hello guys,
So i am using your library I have questions regarding multi-part message. But i want to add BIOREAD and BIOMATCH command i have added them in my local environment but as by default message buffer size is of only 256 bytes how to send more then 256 bytes in chunks i have read document from SIA version 2.2 then have given some insite aboute multi-part message but how to do it in any specific command?

**Comment by sidcha** [comment]
Hello! There is some form of multi-part message support in LibOSDP with respect to file transfer operation; but not for general support for multi-part messaging. Can you post a link to how you have done the BIOREAD and BIOMATCH commands? then we can see how it can be incorporated into the existing design.

**Reply by jaldeepkaneria** [reply]
Hello! so Basically I need  BIOREAD and BIOMATCH Command For Finger print. I have added BIOREAD command till now. Mainly I have made only basic changes in files like osdp.h, osdp_cp.c and osdp_pd.c file which I have attached in package. I have basically added command in osdp_cmd_e for BIOREAD and BIOMATCH and handle in send command related function. Also for PD Part i have received command in pd_decode_command() function After and Provide proper reply id and build reply pd_build_reply() function I have Scanning functionality for finger print in my application code which i can not disclose because of my firms rules in which I work. I have modified OSDP_CMD_RETRY_WAIT_MS, and OSDP_PACKET_BUF_SIZE to receive reply in single message as i can't find solution for multi-part message so i wanted to know if we can build multi-part message then I can handle various biometric type in reply and provide data in CP in message chunks. Also Biometric data can be very as per hardware and other specification so one can not take any fixed buffer size also if buffer size taken is very large then it will may take unnecessary time over head in each message.
[libosdp-2.2.0.zip](https://github.com/goToMain/libosdp/files/14591930/libosdp-2.2.0.zip)

**Reply by sidcha** [reply, marked as answer]
I took a quick look at the multi-part message specification. They specified the structure of the multi-part message but failed to say where this has to be placed in the packet structure. Then there are also other commands that have their own definition of where and how the multi-part management has to be done.

For this to be supported generically (any message that exceeds the packet size can request it to be sent as multi-part), the multi-part fragment info has to be part of the header. And unfortunately this is not the case.
